src/psyclone/dynamo0p3.py

```diff
@@ -8831,6 +8837,132 @@ def stencil(self, value):
+                root_table.add(fld_type)
+            return fld_type
+
+        if self.is_scalar:
+
+            api_config = Config.get().api_conf("dynamo0.3")
+
+            if self.intrinsic_type == 'real':
+                kind_name = api_config.default_kind["real"]
+                prim_type = ScalarType.Intrinsic.REAL
+            elif self.intrinsic_type == 'integer':
+                kind_name = api_config.default_kind["integer"]
+                prim_type = ScalarType.Intrinsic.INTEGER
+            elif self.intrinsic_type == 'logical':
+                kind_name = api_config.default_kind["logical"]
+                prim_type = ScalarType.Intrinsic.BOOLEAN
+            else:
+                raise NotImplementedError(
+                    "Unsupported scalar type '{0}'".format(
+                        self.intrinsic_type))
+            try:
+                kind_symbol = symbol_table.lookup(kind_name)
+            except KeyError:
+                try:
+                    constants_container = symbol_table.lookup(
+                        "constants_mod")
+                except KeyError:
+                    constants_container = ContainerSymbol("constants_mod")
+                    root_table.add(constants_container)
+                kind_symbol = DataSymbol(
+                        kind_name, INTEGER_SINGLE_TYPE,
+                        interface=GlobalInterface(constants_container))
+                root_table.add(kind_symbol)
+            return ScalarType(prim_type, kind_symbol)
+
+        elif self.is_field:
+
+            # Find or create the TypeSymbol for the appropriate field type.
+            if self.intrinsic_type == 'real':
+                mod_name = "field_mod"
+                type_name = "field{0}_type".format(proxy_str)
+            elif self.intrinsic_type == 'integer':
+                mod_name = "integer_field_mod"
+                type_name = "integer_field{0}_type".format(proxy_str)
+            else:
+                raise NotImplementedError(
+                    "Fields may only be of 'real' or 'integer' type but found "
+                    "'{0}'".format(self.intrinsic_type))
+
+            return _find_or_create_type(mod_name, type_name)
+
+        elif self.is_operator:
+
+            # Find or create the TypeSymbol for the appropriate operator type.
+            if self.argument_type == "gh_operator":
+                type_name = "operator{0}_type".format(proxy_str)
+            elif self.argument_type == "gh_columnwise_operator":
+                type_name = "columnwise_operator{0}_type".format(proxy_str)
+            else:
+                raise NotImplementedError(
+                    "Operators may only be of 'gh_operator' or 'gh_columnwise_"
+                    "operator' type but found '{0}'".format(
+                        self.argument_type))
+
+            return _find_or_create_type("operator_mod", type_name)
+
+        else:
+            return DeferredType()
```

Comment:
Do we need/use this, or should it be an exception?

Reply:
That's a good point - I've changed it (and the associated test) to raise/expect an exception.